### PR TITLE
User Story 10-12

### PR DESCRIPTION
**Pull request by sagemlee**
- Add default value to user table column role
-Add flash message when user successfully registers 
-Add error message when user unsuccessfully registers
-Users info must be unique when registering.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,5 @@ coverage
 /yarn-error.log
 
 .byebug_history
+
+/coverage
```
Comment (muydanny) [inline .gitignore:22]: love coverage

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      flash.now[:notice] = new_user.errors.full_messages
+      render :new
+    end
+  end
+
+  def show
+    @user = User.find(session[:user_id])
+  end
+
+  private
```
Comment (stellakunzang) [inline app/controllers/users_controller.rb:20]: Nice, love those strong params!

File: spec/features/user/create_spec.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    expect(current_path).to eq("/profile")
+    expect(page).to have_content("Welcome, #{:name}!")
+  end
+
+
+  xit "will not create user with incorrect info" do
+
+    email =  "georgef@gmail.com"
+    password = "grillingiscool"
```
Comment (muydanny) [inline spec/features/user/create_spec.rb:36]: is it though?

File: spec/features/user/create_spec.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    fill_in :name, with: "George"
+    fill_in :street_address, with: "3 Main st."
+    fill_in :city, with: "Boston"
+    fill_in :state, with: "MA"
+    fill_in :zipcode, with: "12345"
+    fill_in :email, with: email
+    fill_in :password, with: bad_password
+    fill_in :confirm_password, with: bad_password
+    click_on "Create User"
```
Comment (muydanny) [inline spec/features/user/create_spec.rb:44]: Is all this fun stuff in the seed file?

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -1,6 +1,6 @@
 require 'rails_helper'
 
-describe Item, type: :model do
+describe User, type: :model do
```
Comment (muydanny) [inline spec/models/user_spec.rb:3]: nice

File: db/schema.rb [hunk 6]
```diff
@@ -75,7 +75,7 @@
     t.string "zip"
     t.string "email"
     t.string "password_digest"
-    t.integer "role"
+    t.integer "role", default: 0
```
Comment (stellakunzang) [inline db/schema.rb:78]: Woohoo, that's teamwork!

File: app/views/users/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,17 @@
+<%= form_tag '/users', method: :post do %>
```
Comment (muydanny) [inline app/views/users/new.html.erb:1]: this looks nice

**Review by muydanny — APPROVED** [review]

**Review by stellakunzang — APPROVED** [review]
Nice work Super Sage!

File: spec/features/user/create_spec.rb (hunk 3 above)
Comment (stellakunzang) [reply spec/features/user/create_spec.rb:36]: You wanna start a fight with George Foreman?

**Pull request merged**